docs/guide/cli/migrate.md

```diff
@@ -0,0 +1,397 @@
+# migrate
+
+The `migrate` command helps you automatically migrate from packages to more performant alternatives. It can replace packages in your codebase with faster, smaller, or more modern alternatives.
+
+## Overview
+
+The migrate command uses an **opinionated codemod** approach based on e18e's curated list of performance-focused package replacements. This ensures consistent, well-tested migrations across the ecosystem.
+
+The migrate command:
+
+- Identifies packages that can be replaced with better alternatives
+- Automatically updates import statements and package.json
+- Supports both interactive and batch modes
+- Provides dry-run functionality to preview changes
+- Allows custom migration rules via manifest files
+
+## Usage
+
+```sh
+e18e-cli migrate [packages...] [options]
+```
+
+### Arguments
+
+- `packages...` - Names of packages to migrate (e.g., `chalk`, `lodash`)
+
+### Options
+
+| Option | Type | Default | Description |
+|--------|------|---------|-------------|
+| `--all` | `boolean` | `false` | Run all available migrations |
+| `--dry-run` | `boolean` | `false` | Don't apply any fixes, only show what would change |
+| `--include` | `string` | `**/*.{ts,js}` | Files to migrate (glob pattern) |
+| `--interactive` | `boolean` | `false` | Run in interactive mode |
+| `--manifest` | `string[]` | `[]` | Path(s) to custom migration manifest file(s) |
+
+## Examples
+
+### Basic Migration
+
+```sh
+# Migrate a single package
+e18e-cli migrate chalk
+
+# Migrate multiple packages
+e18e-cli migrate chalk lodash moment
+
+# Migrate all available packages
+e18e-cli migrate --all
+```
+
+### Advanced Options
+
+```sh
+# Preview changes without applying them
+e18e-cli migrate chalk --dry-run
+
+# Interactive mode to select packages
+e18e-cli migrate --interactive
+
+# Migrate specific file types
+e18e-cli migrate chalk --include "src/**/*.{ts,js,jsx,tsx}"
+
+# Combine options
+e18e-cli migrate --all --dry-run --include "src/**/*.ts"
+
+# Use custom migration manifest
+e18e-cli migrate --manifest ./custom-migrations.json
+```
+
+## Migration System
+
+### Opinionated Codemod
+
+The CLI uses e18e's **opinionated codemod** approach, which provides a curated list of well-tested, performance-focused package replacements. This ensures consistent migrations across the ecosystem and reduces the risk of breaking changes.
+
+> [!IMPORTANT]
+> The replacement list is still being developed and expanded. The current list represents the initial set of well-tested migrations. More replacements will be added as they are validated and tested across the ecosystem.
+
+### Default Migrations
```

Comment:
this section can go away too

we can't maintain this list here, it'd be too much effort

just link the user to the existing module replacement docs instead (on this website)